-1

**Failed Tests**:   Build  HeaderConsistency 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-1de857/28233/summary.html
**COMMIT**: 78dd13ce1dd7ed33aa6dda9da8689a84d78211ec
**CMSSW**: CMSSW_12_6_X_2022-10-13-1100/el8_amd64_gcc10
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/39716/28233/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_6_X_2022-10-13-1100/src/RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerGlobal.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_6_X_2022-10-13-1100/src/RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerMaxSample.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_6_X_2022-10-13-1100/src/RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerMultiFit.cc
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_6_X_2022-10-13-1100/src/RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerRatio.cc
In file included from /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_6_X_2022-10-13-1100/src/RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerAnalFit.cc:28:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_6_X_2022-10-13-1100/src/RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h:23:23: error: 'Minuit2' is not a namespace-name; did you mean 'Minuit'?
   23 | using namespace ROOT::Minuit2;
      |                       ^~~~~~~
      |                       Minuit
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_6_X_2022-10-13-1100/src/RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerWeights.cc
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_6_X_2022-10-13-1100/src/RecoLocalCalo/EcalRecProducers/plugins/AmplitudeComputationCommonKernels.cu
</pre>

--- a/RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h
+++ b/RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h
@@ -15,9 +15,10 @@
 #include <string>
 
 #include "TROOT.h"
-#include "TMinuit.h"
 #include "TGraph.h"
 #include "TF1.h"
+
+#include "TMinuitMinimizer.h"
 
 template <class C>
 class EcalUncalibRecHitRecAnalFitAlgo : public EcalUncalibRecHitRecAbsAlgo<C> {
@@ -39,6 +40,12 @@ private:
   };
 
 public:
+  EcalUncalibRecHitRecAnalFitAlgo() {
+    //In order to make fitting ROOT histograms thread safe
+    // one must call this undocumented function
+    TMinuitMinimizer::UseStaticMinuit(false);
+  }
+
   // destructor
   ~EcalUncalibRecHitRecAnalFitAlgo<C>() override{};
 
@@ -89,13 +96,12 @@ public:
     TGraph graph(10, xarray, frame);
 
     // fit functions
-    TF1 pulseShape =
-        TF1("pulseShape", "[0]*pow((x - [3])/[1],[2])*exp(-[2]*(x - [1] - [3])/[1])", imax - 1., imax + 3.);
-    TF1 pedestal = TF1("pedestal", "[0]", 0., 2.);
-
-    //TF1 pulseShape = TF1("pulseShape",pulseShapeFunction,imax-1.,imax+3.);
-    //TF1 pedestal = TF1("pedestal",pedestalFunction,0.,2.);
-    TF1 pluseAndPed = TF1("pulseAndPed", "pedestal+pulseShape");
+    TF1 pulseShape = TF1("pulseShape",
+                         "[0]*pow((x - [3])/[1],[2])*exp(-[2]*(x - [1] - [3])/[1])",
+                         imax - 1.,
+                         imax + 3.,
+                         TF1::EAddToList::kNo);
+    TF1 pedestal = TF1("pedestal", "[0]", 0., 2., TF1::EAddToList::kNo);
 
     //pulseShape parameters
     // Amplitude
@@ -119,14 +125,12 @@ public:
     pedestal.SetParameter(0, frame[0]);
     pedestal.SetParName(0, "Pedestal");
 
-    graph.Fit(&pulseShape, "QRM");
-    //TF1 *pulseShape2=graph.GetFunction("pulseShape");
+    int result = graph.Fit(&pulseShape, "QRMN SERIAL");
 
-    if (std::string(gMinuit->fCstatu.Data()) == std::string("CONVERGED ")) {
+    if (0 == result) {
       double amplitude_value = pulseShape.GetParameter(0);
 
-      graph.Fit(&pedestal, "QRL");
-      //TF1 *pedestal2=graph.GetFunction("pedestal");
+      graph.Fit(&pedestal, "QRLN SERIAL");
       double pedestal_value = pedestal.GetParameter(0);
 
       if (!iGainSwitch)

